adam/perception/high_level_semantics_situation_to_developmental_primitive_perception.py

```diff
@@ -156,12 +174,30 @@ def _perceive_property_assertions(self) -> None:
                     perceived_object = self._objects_to_perceptions[situation_object]
                     # Convert the property (which as an OntologyNode object) into PropertyPerception object
                     if BINARY in attributes_of_property:
-                        perceived_property = HasBinaryProperty(
+                        perceived_property: PropertyPerception = HasBinaryProperty(
                             perceived_object, property_
                         )
                     elif COLOR in attributes_of_property:
-                        # TODO: issue: generate perception of colors
-                        raise RuntimeError("Fix this issue COLOR property")
+                        # Sample an RGB value for the color property and generate perception for it
+                        # TODO: Color perception is currently phase-1-ontology-specific, should be changed
+                        if property_ == RED:
```

Comment:
Now that we have five of these and may add more, can you replace this with a map lookup? (e.g. in the ontology Python module, have

```python
COLOR_TO_RGBs = {
   RED : [.....],
  BLUE: [....],
 ....
}
```

then here you can just do

```
r, g, b = self._chooser.choice(COLORS_TO_RDFS[property_)
```

Reply:
Done.